### PR TITLE
Docs: Replace HTTP broken links to equivalent HTTPS resources

### PR DESCRIPTION
**Pull request by sidshank**
## TL;DR ⏱

This PR eliminates some broken HTTP links in *.qhelp files. The PR focuses specifically on `qhelp` files to ensure the changes are low-risk doc changes. 

## Context 🎨

This PR takes a (very) small step towards eliminating some of the http links throughout this repo, as mentioned in [this issue](https://github.com/github/codeql/issues/5163). Before we do a simple search and replace, I thought a good start would be to remove the broken (HTTP status 403 / 404) links.

## How to Review 👀

Just make sure the new links work, and are appropriate for the context in which they are being used. Best reviewed one commit at a time 🍻 🍕 

### grep + sort to find all unique http links in `qhelp` files 🔬

```sh
grep -Eo "http://[a-zA-Z0-9./?=_-]*" -r --no-filename --include \*.qhelp . | sort -u > httpUrls.txt
```

### Quick / Hacky NodeJS script to dump the invalid URLs into a file 🧑‍💻

```js
import fetch, { AbortError } from 'node-fetch';
import fs from 'fs';

const AbortController = globalThis.AbortController;

async function findInvalidUrls() {
    var file = fs.readFileSync('./httpUrls.txt', 'utf8');
    var lines = file.split('\n');
    
    var errorUrlsFile = fs.createWriteStream('./invalidHttpUrls.txt');
    errorUrlsFile.on('error', function (err) {
        console.log(err);
    });

    for (var i = 0; i < lines.length; i++) {
        const controller = new AbortController();
        const timeout = setTimeout(() => {
            controller.abort();
        }, 10000);
        var line = lines[i];
        try {
            const response = await fetch(line, { signal: controller.signal });
            console.log(line + " " + response.status);
            if (response.status >= 400) {
                errorUrlsFile.write(`${line} | ${response.status}\n`);
            }
        } catch (e) {
            if (
                e.code === 'ERR_INVALID_URL' ||
                e.code === 'ENOTFOUND' ||
                e.code === 'ECONNREFUSED'
            ) {
                console.log('At index ' + i + ', ' + line + ' is not a valid url');
            } else if (e instanceof AbortError) {
                console.log('At index ' + i + ', ' + line + ' - Request aborted');
            }
        } finally {
            clearTimeout(timeout);
        }
    }
    errorUrlsFile.end();
}

findInvalidUrls();
```

### Script output 🚰
[invalidHttpUrls.txt](https://github.com/github/codeql/files/9339242/invalidHttpUrls.txt)

Some were false positives, but the rest were indeed bad URLs. I found the appropriate replacements and have made the changes.

### CODE REVIEW

**Review by erik-krogh — APPROVED** [review]
I went through all the new links, and they all look great 👍

**Review by tausbn — APPROVED** [review]
Python bits LGTM! :+1:

**Review by jketema — APPROVED** [review]
C++ 👍

**Review by saritai — APPROVED** [review]
👋 Docs content first responder here! I haven't worked on the codeQL docs in awhile so I forgot how to do a preview of the site, but since this seems quite a simple update, I just tried all the URLs and they work, so going ahead and approving this!

**Review by atorralba — APPROVED** [review]
Java URLs looking good too 👍
